core/domain/topic_domain.py

```diff
@@ -863,6 +863,7 @@ def update_name(self, new_name):
             new_name: str. The updated name for the topic.
         """
         self.name = new_name
+        self.canonical_name = new_name.lower()
```

Comment:
I'm a bit confused ... why is this change in this PR? It seems to be related to the topic name rather than the RTE component.

Reply:
Oh sorry, read your description. If you want this code to go in, could you please make sure this is covered by tests?

Reply:
Done

Reply:
I don't see any additions to topic_domain_test...

Reply:
This function was tested in `topic_services_test`, so added a test there to check this

Reply:
Generally functions in a module should be unit-tested by a test file corresponding to that module though. They may be tested auxiliarily by other files but there should be focused unit tests in topic_domain_test.py, right?

Reply:
Yeah, but the `apply_change` function in topic_services calls all these update functions. So, the tests for all the update functions are in topic_services.

Reply:
Oh ok. Technically we should be creating a dummy topic in topic_domain_test.py and applying these functions on it.

But if existing code has that format let's follow that structure. We should probably revamp it in the future though.